roles/openshift_logging_fluentd/tasks/main.yaml

```diff
@@ -0,0 +1,258 @@
+
+- set_fact: tempdir="{{ openshift_logging_tempdir }}"
+  when: openshift_logging_tempdir
+
+- name: Copy the admin client config(s)
+  command: >
+    cp {{ openshift_master_config_dir }}/admin.kubeconfig {{ tempdir }}/admin.kubeconfig
+  changed_when: False
+  check_mode: no
+  when: not openshift_logging_tempdir or openshift_logging_tempdir == ''
+
+- name: Create templates subdirectory
+  file:
+    state: directory
+    path: "{{ tempdir }}/templates"
+    mode: 0755
+  changed_when: False
+  check_mode: no
+  when: not openshift_logging_tempdir or openshift_logging_tempdir == ''
+
+# we want to make sure we have all the necessary components here
+
+# create service account
+- name: Create Fluentd service account
+  oc_serviceaccount:
+    state: present
+    kubeconfig: "{{ tempdir }}/admin.kubeconfig"
+    name: "aggregated-logging-fluentd"
+    namespace: "{{ openshift_logging_namespace }}"
+    image_pull_secrets: "{{ openshift_logging_image_pull_secret }}"
+  when: openshift_logging_image_pull_secret != ''
+
+- name: Create Fluentd service account
+  oc_serviceaccount:
+    state: present
+    kubeconfig: "{{ tempdir }}/admin.kubeconfig"
+    name: "aggregated-logging-fluentd"
+    namespace: "{{ openshift_logging_namespace }}"
+  when:
+  - openshift_logging_image_pull_secret == ''
+
+# set service account scc
+- name: Set privileged permissions for Fluentd
+  oc_adm_policy_user:
+    kubeconfig: "{{ tempdir }}/admin.kubeconfig"
+    namespace: "{{ openshift_logging_namespace }}"
+    resource_kind: scc
+    resource_name: privileged
+    state: present
+    user: "system:serviceaccount:{{ openshift_logging_namespace }}:aggregated-logging-fluentd"
+
+# set service account permissions
+- name: Set cluster-reader permissions for Fluentd
+  oc_adm_policy_user:
+    kubeconfig: "{{ tempdir }}/admin.kubeconfig"
+    namespace: "{{ openshift_logging_namespace }}"
+    resource_kind: cluster-role
+    resource_name: cluster-reader
+    state: present
+    user: "system:serviceaccount:{{ openshift_logging_namespace }}:aggregated-logging-fluentd"
+
+# create Fluentd configmap
+- block:
+    - template:
+        src: "{{ fluentd_version }}/fluent.conf.j2"
+        dest: "{{ tempdir }}/fluent.conf"
+      vars:
+        deploy_type: "{{ openshift_logging_fluentd_deployment_type }}"
+      when: fluentd_config_contents is undefined
+      changed_when: no
+
+    - copy:
+        src: "{{ fluentd_version }}/fluentd-throttle-config.yaml"
+        dest: "{{ tempdir }}/fluentd-throttle-config.yaml"
+      when: fluentd_throttle_contents is undefined
+      changed_when: no
+
+    - copy:
+        src: "{{ fluentd_version }}/secure-forward.conf"
+        dest: "{{ tempdir }}/secure-forward.conf"
+      when: fluentd_securefoward_contents is undefined
+
+      changed_when: no
+
+    - copy:
+        content: "{{ fluentd_config_contents }}"
+        dest: "{{ tempdir }}/fluent.conf"
+      when: fluentd_config_contents is defined
```

Comment:
I'll make sure its doing what I expect, but it should be that if you provide a fluentd config in that variable we will use that instead of the template provided one